src/net_processing.cpp

```diff
@@ -2401,38 +2405,39 @@ arith_uint256 PeerManagerImpl::GetAntiDoSWorkThreshold()
  *
  * We'll send a getheaders message in response to try to connect the chain.
  *
- * The peer can send up to MAX_UNCONNECTING_HEADERS in a row that
+ * The peer can send up to MAX_UNCONNECTING_HEADERS_MSGS in a row that
  * don't connect before given DoS points.
  *
  * Once a headers message is received that is valid and does connect,
- * nUnconnectingHeaders gets reset back to 0.
+ * m_unconnecting_headers_msgs gets reset back to 0.
  */
 void PeerManagerImpl::HandleFewUnconnectingHeaders(CNode& pfrom, Peer& peer,
         const std::vector<CBlockHeader>& headers)
 {
     const CNetMsgMaker msgMaker(pfrom.GetCommonVersion());
 
-    LOCK(cs_main);
-    CNodeState *nodestate = State(pfrom.GetId());
-
-    nodestate->nUnconnectingHeaders++;
+    peer.m_unconnecting_headers_msgs++;
     // Try to fill in the missing headers.
     if (MaybeSendGetHeaders(pfrom, GetLocator(m_chainman.m_best_header), peer)) {
-        LogPrint(BCLog::NET, "received header %s: missing prev block %s, sending getheaders (%d) to end (peer=%d, nUnconnectingHeaders=%d)\n",
+        LogPrint(BCLog::NET, "received header %s: missing prev block %s, sending getheaders (%d) to end (peer=%d, m_unconnecting_headers_msgs=%d)\n",
             headers[0].GetHash().ToString(),
             headers[0].hashPrevBlock.ToString(),
             m_chainman.m_best_header->nHeight,
```

Comment:
Since this is no longer locked with `cs_main`, I'm wondering if the `m_chainman.m_best_header` passed to `MaybeSendGetHeaders` and the `m_chainman.m_best_header->nHeight` passed to `LogPrint` could be different? I know it's just logging and not super critical, but could be annoying for debugging if so. Probably better to make that more robust?

Reply:
Added a commit to annotate `m_best_header` as guarded by `cs_main` and fixed this up to log the same height that is being passed to `MaybeSendGetHeaders`.